src/main.py

```diff
@@ -225,9 +239,26 @@ def commit_changes(changes: t.List[str]) -> CommittedChanges:
     if not base_branch:
         base_branch = None
 
-    # Moved to detached HEAD
-    log.info("Switching to detached HEAD")
-    check_call(["git", "-c", "advice.detachedHead=false", "checkout", "HEAD@{0}"])
+    if target_branch:
+        log.info("Committing directly to existing branch '%s'", target_branch)
+        if not branch_exists(target_branch):
+            log.error("Branch '%s' does not exist locally", target_branch)
+            return None
+        check_call(["git", "checkout", target_branch])
```

Comment:
Running `git checkout` at this point may fail. By this point there are uncommitted changes in the working copy. Suppose you are on branch `main`, and you say `--commit-branch test`. If `test` exists but there are divergent changes that conflict with the uncommitted changes, then `git checkout test` will fail.

I think it would be simpler for this option to not take an argument. Instead you would say `--dont-switch-branch` (name tbd) and it would commit directly to whatever branch happens to be checked out, and not create a new one.